### ISSUE TITLE
Notifications not delivered for user pings in comment threads

### ISSUE THREAD

**Issue by cellio** [codidact/qpixel#1475]
meta:291852

A user made a post and is thus automatically subscribed to comment threads.  The user received notifications for new comments *except* if those comments included @-mentions of the user -- exactly when we would want to prioritize a ping.  We did some testing (see the comment thread on that meta question), and the user did not receive *any* ping for a new comment *if* it pinged that user.  Ping-less comments continued to deliver notifications.

**Comment by trichoplax** [comment]
Do we have a defined list of things that pings should be expected to do?

If you are following a thread (either automatically through being the author of the thread or post, or due to manually following) then you should be notified of every comment. If you unfollow a thread then you should not be notified of any comments. If you have never followed a thread you should not be notified of any comments.

Is there reason to expect mentioning a user to override any of those behaviours? If I am already being notified, presumably the mention should have no further effect, and is simply to inform anyone reading the thread that the comment is aimed at me? If I unfollow, should a mention be able to override that? I assume a mention cannot ping someone who has no relationship with the thread (has never followed), but is there the expectation that people who unfollow can still be reached with a mention?

It seems it might be useful (whether in this issue or another) to make 2 things explicit:
- When a user unfollows, inform them that they will / will not be notified when someone mentions them.
- When someone mentions a user who has unfollowed, inform the mentioner that the mentionee will / will not be notified by it. This might make a user's following status sometimes public - is this a concern?

**Comment by cellio** [comment]
I agree: if you're following, you should receive all notifications.  It sounds like we have a bug where some are not delivered.

If you've never participated in a thread (and aren't the post owner), you're not supposed to be pingable.  I don't know what happens if you would have been pingable but you unfollowed -- does your name show up in red if someone tries to ping you, like would be the case for an uninvolved user?  Something to investigate.  I agree this reveals your following status and that might not be desirable.  Sounds like we need to work out what behavior we want for the unfollow case and then maybe make changes. Meanwhile, we seem to have a bug in the follow case.

**Comment by trichoplax** [comment]
Yes I agree the specific bug raised here should be fixed without waiting for the discussion around people who have unfollowed. I just wanted to quickly mention it while I thought of it, but this might not be the right place.

**Comment by cellio** [comment]
It's a good catch, one I hadn't thought about before -- thanks for raising it!  Now to make sure it doesn't get lost...

**Comment by trichoplax** [comment]
Hopefully that should help keep track of it.
